### ISSUE TITLE
Add a logging abstraction to the Tracer

### ISSUE THREAD

**Issue by austinlparker** [lightstep/lightstep-tracer-csharp#37]
Right now the only way to really figure out what's going on in the Tracer is to attach a debugger. We should also be emitting logs through some sort of abstraction layer to make it easier to figure out what's going on.

**Comment by austinlparker** [comment]
This is blocked by https://github.com/damianh/LibLog/issues/189

**Comment by austinlparker** [comment]
In the meantime, I can build LibLog locally with the right targets and start to add logging, this is where I'm at in terms of debug level stuff.

```[09:07:15 DBG] Creating new tracer with GUID 5804151968470634613. Project Access Token: <redacted>, Report Period: 00:00:05, Report Timeout: 00:00:30.
[09:07:15 DBG] Tracer is reporting to https://collector.lightstep.com:443/api/v2/reports.
[09:07:15 DBG] Flushing SpanData
[09:07:15 DBG] Lock freed, getting current buffer.
[09:07:15 DBG] 0 spans copied from buffer.
[09:07:15 DBG] Serializing spans to proto.
[09:07:16 DBG] Serialization complete in 251ms. Request size: 432b.
[09:07:16 DBG] Report HTTP Response OK
[09:07:20 DBG] Flushing SpanData
[09:07:20 DBG] Lock freed, getting current buffer.
[09:07:20 DBG] 384 spans copied from buffer.
[09:07:20 DBG] Serializing spans to proto.
[09:07:25 DBG] Flushing SpanData
[09:07:25 DBG] Lock freed, getting current buffer.
[09:07:25 DBG] 194 spans copied from buffer.
[09:07:25 DBG] Serializing spans to proto.
[09:07:30 DBG] Flushing SpanData
[09:07:30 DBG] Lock freed, getting current buffer.
[09:07:30 DBG] 136 spans copied from buffer.
[09:07:30 DBG] Serializing spans to proto.
[09:07:35 DBG] Flushing SpanData
[09:07:35 DBG] Lock freed, getting current buffer.
[09:07:35 DBG] 112 spans copied from buffer.
[09:07:35 DBG] Serializing spans to proto.
[09:07:39 DBG] Serialization complete in 12539ms. Request size: 23147b.
[09:07:39 DBG] Serialization complete in 18221ms. Request size: 45224b.
[09:07:39 DBG] Report HTTP Response OK
[09:07:40 DBG] Report HTTP Response OK
[09:07:40 DBG] Serialization complete in 9490ms. Request size: 16365b.
[09:07:40 DBG] Report HTTP Response OK
[09:07:40 DBG] Flushing SpanData
[09:07:40 DBG] Lock freed, getting current buffer.
[09:07:40 DBG] 118 spans copied from buffer.
[09:07:41 DBG] Serializing spans to proto.
[09:07:42 DBG] Serialization complete in 6466ms. Request size: 13565b.
[09:07:42 DBG] Report HTTP Response OK
[09:07:42 DBG] Flushing SpanData
[09:07:42 DBG] Lock freed, getting current buffer.
[09:07:43 DBG] 56 spans copied from buffer.
[09:07:43 DBG] Serializing spans to proto.
[09:07:45 DBG] Flushing SpanData
[09:07:45 DBG] Lock freed, getting current buffer.
[09:07:45 DBG] 0 spans copied from buffer.
[09:07:45 DBG] Serializing spans to proto.
[09:07:46 DBG] Serialization complete in 360ms. Request size: 432b.
[09:07:46 DBG] Report HTTP Response OK
[09:07:46 DBG] Serialization complete in 5511ms. Request size: 14280b.
[09:07:46 DBG] Serialization complete in 3398ms. Request size: 6987b.
[09:07:46 DBG] Report HTTP Response OK
[09:07:46 DBG] Report HTTP Response OK
[09:07:50 DBG] Flushing SpanData
[09:07:50 DBG] Lock freed, getting current buffer.
[09:07:50 DBG] 0 spans copied from buffer.
[09:07:50 DBG] Serializing spans to proto.
[09:07:50 DBG] Serialization complete in 160ms. Request size: 432b.
[09:07:50 DBG] Report HTTP Response OK
```

**Comment by austinlparker** [comment]
fixed via #44
